[Custom Amounts M2] Re-structure payment section of order creation

toupper opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #10884 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
With this PR we restructure the Payment section of order creation to match the new design ichqhxDjbmZZmvJaJkGhkj-fi-118%3A6643

As this goes directly into production, we don't need feature flag.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

1. Go to Orders
2. Tap on + to create a new order.
3. Check that the new design matches:

- Empty State (no items added)
- Add products, custom amounts, shipping, coupons, discounts, gift cards to check that it works fine.
- Add customer details and notes to check that it works according to design as well.
- Check that when the order is not editable (e.g. completed) 

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

https://github.com/woocommerce/woocommerce-ios/assets/1864060/a8080cb5-7ae0-4b60-9dff-e44c6a829dd6

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

toupper committed:
WIP: Order creation - payment details empty order state

toupper committed:
Add customer data logic for rendering empty state

toupper committed:
Restructure order payment section when products or custom amounts are added

toupper committed:
Add products row

toupper committed:
Adjust color according to design

toupper committed:
Add missing divider

toupper committed:
Remove commented code

toupper committed:
Add tests for the paymentDataViewModel

toupper committed:
Add release note

toupper committed:
Update strings

iamgabrielma reviewed (commented):
Looks and works good! With the re-structure there's an UI issue when "Coupons tooltip" is visible, since this appears behind "Set New Tax rate":
<img width=600 src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/4b396250-bf09-4489-a4c2-6b874d3a8557">

In order to reproduce it: Add a product to an order > add a discount to the product > this enables the coupons tooltip > now tap on `(?)`

There's also a difference in the color of shipping when an order is non-editable, I'm not sure if this is expected since the shipping row is clickable by itself as a link, so could be expected that's the default style:

<img width=600 src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/f3319020-697a-4970-9803-34ba53d21493">

toupper commented:
Thanks for your review and added comment @iamgabrielma !

I added an issue https://github.com/woocommerce/woocommerce-ios/issues/11049 for the first remark, as it's out of the scope of this PR. The second one is fixed with this https://github.com/woocommerce/woocommerce-ios/pull/11048 for more convenience.

toupper committed:
Merge branch 'trunk' into issue/10884-custom-amounts-m2-rest-of-order-creation
